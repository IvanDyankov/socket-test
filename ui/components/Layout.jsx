import React from 'react';
import Grid from '@material-ui/core/Grid';
import NowOnline from './NowOnline';
import MainChat from './MainChat';
import MessagePanel from './MessagePanel';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  root: {
    backgroundColor: 'lightgrey'
  }
});

class Layout extends React.Component {

  render() {
    const { classes } = this.props;
    return (
      <div className={classes.root}>
        <Grid container className={classes.main} justify="center" direction="row">
          <Grid item xs="3" >
            <NowOnline />
          </Grid>
          <Grid item xs="9">
            <Grid container direction="column">
              <MainChat />
              <MessagePanel />
            </Grid>
          </Grid>
        </Grid>
      </div>
    );
  }
};

export default withStyles(styles)(Layout);

