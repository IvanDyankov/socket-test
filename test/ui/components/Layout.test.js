import React from 'react';
import Layout from '../../../ui/components/Layout';
import { shallow } from 'enzyme';

describe('<Layout />', () => {

  it('should render correctly with default props', () => {
    const wrapper = shallow(<Layout />);
    expect(wrapper).toMatchSnapshot();
  });
});
