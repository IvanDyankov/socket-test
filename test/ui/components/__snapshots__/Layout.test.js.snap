// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render correctly with default props 1`] = `
<Layout
  classes={
    Object {
      "root": "Layout-root-1",
    }
  }
/>
`;
